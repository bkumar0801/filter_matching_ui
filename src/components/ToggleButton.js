import React from 'react';

class ToggleButton extends React.Component {
  constructor({ initialChecked }) {
    super();
    this.state = { checked: initialChecked }
  }
  onTextChanged() {
    const newState = !this.state.checked;
    this.setState({ checked: newState }); // we update our state
    this.props.callbackParent(newState); // we notify our parent
  }
  render() {
    return <label>
    {this.props.text}: <input type="checkbox"
    checked={this.state.checked}
    onChange={() => this.onTextChanged()}/>
    </label>
  }
}

export default ToggleButton;