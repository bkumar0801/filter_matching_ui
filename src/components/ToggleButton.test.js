import React from 'react';
import { mount, shallow, enzyme } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {sinon, spy, stub} from 'sinon';
import ToggleButton from './ToggleButton';

configure({ adapter: new Adapter() });

describe(ToggleButton, () => {
    const mockOnChange = jest.fn();
    const component = shallow(<ToggleButton initialChecked={true} onChange={mockOnChange}/>);

    it('renders and matches our snapshot', () => {
        const component = renderer.create(<ToggleButton />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("contains input and label", () => {
        expect(component.find("input")).toHaveLength(1);
        expect(component.find("label")).toHaveLength(1);
    });

    it("has checked state true", () => {
        expect(component.state('checked')).toEqual(true);
    });

    it("change checked state of checkbox on setState", () => {
        const wrapper = shallow(<ToggleButton initialChecked={true}/>);
        wrapper.setProps({ checked: true });
        let checkbox = wrapper.find({ type: 'checkbox' });
        expect(checkbox.props().checked).toEqual(true);
        wrapper.setState({ checked: false });
        checkbox = wrapper.find({ type: 'checkbox' });
        expect(checkbox.props().checked).toEqual(false);
    });
});