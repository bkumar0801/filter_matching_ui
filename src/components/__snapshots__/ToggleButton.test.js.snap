// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class ToggleButton extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = { checked: props.initialChecked };
  }
  onStateChanged() {
    const newState = !this.state.checked;
    this.setState({ checked: newState }); // we update our state
    this.props.callbackParent(newState); // we notify our parent
  }
  render() {
    return _react2.default.createElement("label", { __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this },
      this.props.text, ": ", _react2.default.createElement("input", { type: "checkbox",
        checked: this.state.checked,
        onChange: () => this.onStateChanged(), __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this }));

  }} renders and matches our snapshot 1`] = `
<label>
  : 
  <input
    checked={undefined}
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;
